src/js/components/RulePicker.react.js

```diff
@@ -87,10 +88,9 @@ class RulePicker extends React.Component<Props, State> {
               valid: this.props.rule.selector != '',
             })}
           >
-            <label>
-              {this.props.rule.selector != '' ? <span>✔</span> : <span>✘</span>}{' '}
+            <LabelIcon valid={hasSelector} required={!hasSelector}>
```

Comment:
As I said on the component, I think it would be nice to instead of sending valid and require props and the component to handle the logic, to send the actual values we want to display and the component just to handle how it looks and behave.

Reply:
@gisedaye At the beginning I implemented the LabelIcon as a generic component that receives the icon and label for the tooltip but when using the component, the logic, the icon and text were being repeated in different codes, so I added them all together, with that, in case we decide to change the tooltip label or the icon, we don't need to update in more than one place.

But, thinking now, we can use the `LabelIcon` component receiving the icon and tooltip label and create a new file in the `/utils` folder to centralize hardcoded strings, what do you think?

Reply:
Sounds good to me! I would go for either custom components for each state (Valid, Required, Optional) reusing the generic component and/or to centralize texts. I think centralizing texts is good enough for now!